spec/datadog/tracing/context_provider_spec.rb

```diff
@@ -106,59 +106,74 @@ def thread_contexts
   end
 
   describe '#local' do
-    subject(:local) { thread_local_context.local }
+    subject(:local) { fiber_local_context.local }
 
-    context 'with a second ThreadLocalContext' do
-      let(:thread_local_context2) { described_class.new }
+    context 'with a second FiberLocalContext' do
+      let(:fiber_local_context2) { described_class.new }
 
-      it 'does not interfere with other ThreadLocalContext' do
-        local_context = thread_local_context.local
-        local_context2 = thread_local_context2.local
+      it 'does not interfere with other FiberLocalContext' do
+        local_context = fiber_local_context.local
+        local_context2 = fiber_local_context2.local
 
         expect(local_context).to_not eq(local_context2)
-        expect(thread_local_context.local).to eq(local_context)
-        expect(thread_local_context2.local).to eq(local_context2)
+        expect(fiber_local_context.local).to eq(local_context)
+        expect(fiber_local_context2.local).to eq(local_context2)
+      end
+    end
+
+    context 'in another fiber' do
+      it 'create one fiber-local variable per fiber' do
+        context = fiber_local_context.local
+
+        Fiber.new do
+          expect { @fiber_context = fiber_local_context.local }
+            .to change { thread_contexts.size }.from(0).to(1)
```

Comment:
Minor: Should `thread_contexts` be renamed as well?

Reply:
I wasn't 100% sure if it would make sense to rename `thread_contexts`, given the values do come from threads, but they happen to match with the respective fibers.

I ended up renaming it now, take a look :)